core/src/main/java/org/kiwix/kiwixmobile/core/main/KiwixWebView.kt

```diff
@@ -0,0 +1,206 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import android.net.Uri
+import android.os.Environment
+import android.os.Handler
+import android.os.Message
+import android.util.AttributeSet
+import android.util.Log
+import android.view.ContextMenu
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebView
+import com.cprcrack.videowebview.VideoEnabledWebView
+import io.reactivex.disposables.CompositeDisposable
+import org.kiwix.kiwixmobile.core.BuildConfig
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.coreComponent
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.instance
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.extensions.toast
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils.Companion.getCurrentLocale
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import java.io.File
+import java.io.IOException
+import javax.inject.Inject
+
+private const val INITIAL_SCALE = 100
+
+@SuppressLint("ViewConstructor")
+@SuppressWarnings("LongParameterList")
+open class KiwixWebView @SuppressLint("SetJavaScriptEnabled") constructor(
+  context: Context,
+  private val callback: WebViewCallback,
+  attrs: AttributeSet?,
+  nonVideoView: ViewGroup?,
+  videoView: ViewGroup?,
+  webViewClient: CoreWebViewClient?
+) : VideoEnabledWebView(context, attrs) {
+  @Inject
+  lateinit var sharedPreferenceUtil: SharedPreferenceUtil
+
+  @Inject
+  lateinit var zimReaderContainer: ZimReaderContainer
+
+  private val compositeDisposable = CompositeDisposable()
+
+  private fun setWindowVisibility(systemUiVisibility: Int) {
+    (context as Activity).window.decorView.systemUiVisibility = systemUiVisibility
+  }
+
+  init {
+    if (BuildConfig.DEBUG) {
+      WebView.setWebContentsDebuggingEnabled(true)
+    }
+    coreComponent.inject(this)
+    // Set the user agent to the current locale so it can be read with navigator.userAgent
+    settings.apply {
+      userAgentString = "${getCurrentLocale(context)}"
+      domStorageEnabled = true
+      javaScriptEnabled = true
+      loadWithOverviewMode = true
+      useWideViewPort = true
+      setInitialScale(INITIAL_SCALE)
+      builtInZoomControls = true
+      displayZoomControls = false
+      clearCache(true)
+      allowUniversalAccessFromFileURLs = true
+      setWebViewClient(webViewClient)
+    }
+    val client = KiwixWebChromeClient(callback, nonVideoView, videoView, this)
```

Comment:
similarly this client variable need not exist, a rewrite with apply should suffice

Reply:
done